### ISSUE TITLE
Add status and headers methods to ::Response::Delayed

### ISSUE THREAD

**Issue by veryrusty** [PerlDancer/Dancer2#894]
Following on from @szabgab report of breakage on the mailing list after upgrading to 0.160000:
Add `status` and `headers` methods to Dancer2::Core::Response::Delayed.

A delegation (via `handles` on the response attribute) is probably the simplest approach.

**Comment by yanick** [comment]
The thing Gabor said:

http://lists.preshweb.co.uk/pipermail/dancer-users/2015-April/004541.html

```
I have code like this:

hook after => sub {

    my ($response) = @_;
    if ($response->status != 200) {
       ...
    }

    return;

};

In Dancer2-0.159003  the after hooks used to get an object that had a
method called 'status'.

In Dancer2-0.160000 it is not the case any more.

I get the following:  Can't locate object method "status" via package
"Dancer2::Core::Response::Delayed"

```

**Comment by szabgab** [comment]
The solution to my issue was to use the ```response``` function provided by the DSL instead of the parameter that was passed.

If I understood correctly what @xsawyerx indicated in the email ( http://lists.preshweb.co.uk/pipermail/dancer-users/2015-April/004545.html ), that there is no commitment to always pass that parameter to after hook (or for that matter to any hook), then I'd think it is better to not pass any parameter. If you pass a parameter, even undocumented, people will rely on it and when you change it they will blame you for code breakage. IMHO it would be better to avoid that.

If you are worried that people already rely on the object then you might pass an object that will throw an exception and point the user to the explanation how to fix the issue.

**Comment by veryrusty** [comment]
Resolved by #898. Thanks to all involved!  :dancers:
